src/main/java/com/ids1024/whitakerswords/WhitakersWords.java

```diff
@@ -33,60 +34,102 @@
 import android.widget.ListView;
 
 public class WhitakersWords extends ListActivity {
-    String search_term;
-    ArrayList<SpannableStringBuilder> results;
-    ListView list_view;
-    EditText search_term_view;
-    ToggleButton english_to_latin_view;
-
-    public void copyFiles() throws IOException {
-        for (String filename: getAssets().list("words")) {
-            InputStream ins = getAssets().open("words/" + filename);
-            byte[] buffer = new byte[4096];
-            FileOutputStream fos = openFileOutput(filename, MODE_PRIVATE);
-            for (int read = 0; read >= 0; read = ins.read(buffer)) {
+    private static final String WORDS_EXECUTABLE = "words";
+
+    private String search_term;
+    private final ArrayList<SpannableStringBuilder> results = new ArrayList<>();
+    private ListView list_view;
+    private EditText search_term_view;
+    private ToggleButton english_to_latin_view;
+
+    private File getFile(String filename) {
+      return new File(getCacheDir(), filename);
+    }
+
+    private void copyFiles() throws IOException {
+        byte[] buffer = new byte[32*1024];
+        for (String filename : getAssets().list("words")) {
+            copyFile(filename, buffer);
+        }
+
+        getFile(WORDS_EXECUTABLE).setExecutable(true);
+    }
+
+    private void copyFile(String filename, byte[] buffer) throws IOException {
+        InputStream ins = null;
+        FileOutputStream fos = null;
+        File outputFile = getFile(filename);
+        // if the file already exists, don't copy it again 
+        if (outputFile.exists()) {
```

Comment:
Yeah, I had been meaning to make it not extract all the files every time the app runs. However, it needs some way to allow updating these files when the app updates.

Perhaps it should save the current version of whitakers-words-android to a file in the cache directory, and update the files if it differs? Or is there a better way to do this?